### PR TITLE
Test compilation of bad sources

### PR DESCRIPTION
**Pull request by unindented**
Added tests for bad sources, as I saw an issue with the new `compileToString` method incorrectly resolving when it should reject.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -164,6 +164,10 @@ function compileToString(sources, options){
       compile(sources, options)
         .on("close", function(exitCode){
           fs.readFile(info.path, function(err, data){
+            if (exitCode !== 0) {
+              err = "Compiler process exited with code " + exitCode;
+            }
```
Comment (unindented) [inline index.js]: Non-zero exit codes were being ignored, and the promise was being resolved, when it should have been rejected. 😞

Comment (eeue56) [inline index.js]: :+1:

File: test/compile.js [hunk 2]
```diff
@@ -23,13 +23,23 @@ describe("#compile", function() {
...
+
+  it("reports errors on bad source", function (done) {
+    var opts = {yes: true, verbose: true, cwd: fixturesDir};
+    var compileProcess = compiler.compile(prependFixturesDir("Bad.elm"), opts);
+
+    compileProcess.on("exit", function(exitCode) {
+      assert.equal(exitCode, 1, "Expected elm-make to have exit code 1");
+      done();
+    });
+  });
```
Comment (eeue56) [inline test/compile.js]: :+1:

**Comment by eeue56** [comment]
Looks good, merge it!

**Comment by rtfeldman** [comment]
Published as 2.3.3. Thanks @unindented!

**Pull request merged**